libraries/node-core-library/src/Executable.ts

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as child_process from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+
+import { FileSystem, PosixModeBits } from './FileSystem';
+
+/**
+ * Typings for one of the streams inside IExecutableSpawnSyncOptions.stdio.
+ * @beta
+ */
+export type ExecutableStdioStreamMapping = 'pipe' | 'ignore' | 'inherit'
+  | NodeJS.WritableStream | NodeJS.ReadableStream
+  | number | undefined;
+
+/**
+ * Typings for IExecutableSpawnSyncOptions.stdio.
+ * @beta
+ */
+export type ExecutableStdioMapping = 'pipe' | 'ignore' | 'inherit' | ExecutableStdioStreamMapping[];
```

Comment:
>ExecutableStdioStreamMapping [](start = 69, length = 28)

Can you specify that the max number here is 3?

Reply:
Actually any number of entries are allowed.  If it's more than 3, then additional pipes are created. If it's less than 3, then defaults are preserved.

[https://nodejs.org/api/child_process.html#child_process_options_stdio](https://nodejs.org/api/child_process.html#child_process_options_stdio)

---
In reply to: [208417717](https://github.com/Microsoft/web-build-tools/pull/764#discussion_r208417717) [](ancestors = 208417717)